sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -595,6 +595,14 @@ def equal_split(a: SplitPrimitiveTensor, b: AnyTensor) -> bool:
     return a.is_deep_equal(b)
 
 
+@expand.override(ReplicatedTensor)
+def expand_replicated(tensor: ReplicatedTensor, shape: List[int]) -> ReplicatedTensor:
+    assert len(shape) == len(tensor.shape)
```

Comment:
Technically we don't need this assert. If something is wrong the underlying `torch.Tensor.expand` op will fail.
Also this assert restricts some valid expands where new leading dimensions are inserted. Without the assert we would support this for free.